### PR TITLE
[Fix] `forbid-component-props`: Implemented support for "namespaced" components.

### LINKED ISSUE #2766: forbid-component-props: How to distinguish a component from a library?
Situation:

```tsx
import {Layout as AntdLayout} from 'antd';
...
            <AntdLayout.Content className={contentClassNames}>
...
```

Adding `AntdLayout` nor `AntdLayout.Content` into `allowedFor` isn't working. I don't want to add `Content`, because then linter won't report an error when our `Content` component is used with `className`.

Edit: I have attempted to improve it in https://github.com/yannickcr/eslint-plugin-react/pull/2767.

### PR DESCRIPTION
**Pull request by mnn**
I don't really know what I am doing. I have just looked at the structure of the `node`, added a test and updated code so the tests pass. 

Fixes #2766.

### CODE REVIEW

File: lib/rules/forbid-component-props.js [hunk 1]
```diff
@@ -78,7 +78,12 @@ module.exports = {
 
     return {
       JSXAttribute(node) {
-        const tag = node.parent.name.name;
+        const parentName = node.parent.name;
+        let tag = parentName.name;
+        if (!tag && parentName.object.name !== 'this') {
```
Comment (ljharb) [inline lib/rules/forbid-component-props.js:83]: why not `this`? iow, `<this.Foo />` should be valid as well, no?

Comment (mnn) [reply lib/rules/forbid-component-props.js:83]: One test was failing, so I thought there is some reason I am not aware of behind forbidding it.

Comment (ljharb) [reply lib/rules/forbid-component-props.js:83]: Why would it be a different component every time? `this.Component` could just be an own property on the instance, the same every time.

I think it's reasonable to assume that `this` inside a class method is always the class instance.

Comment (ljharb) [reply lib/rules/forbid-component-props.js]: I've taken care of it; it's that the "first letter is capitalized" check wasn't yet taking into account the namespace and the dot.

**Pull request merged**